app.js

```diff
@@ -82,6 +83,36 @@ function updateCode() {
 	prettyPrint();
 }
 
+function readData() {
+	var bytestr = prompt("Input your data here, in hex format: 0x1, 0x2");
+	if (!bytestr) return;
+	var width = matrix.length;
```

Comment:
Width and height should be:
```suggestion
	var width = matrix[0].length;
	var height = matrix.length;
```